src/Npgsql/MultiHostConnectorPoolWrapper.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Npgsql.Util;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Transactions;
+
+namespace Npgsql
+{
+    sealed class MultiHostConnectorPoolWrapper : ConnectorSource
+    {
+        readonly ConnectorSource _wrappedSource;
```

Comment:
This can be a MultiHostConnectorPool directly, and the virtual methods on MultiHostConnectorPool can be sealed - this should help devirtualization; right now we do two virtual dispatches, once for the wrapper and once for the pool.